src/types/index.ts

```diff
@@ -0,0 +1,213 @@
+type TransformedEvent = {
+  version: string;
+  type: string;
+  method: string;
+  endpoint: string;
+  userId: string;
+  headers: Record<string, unknown>;
+  params?: Record<string, unknown>;
+  body?: {
+    JSON?: Record<string, unknown>;
+    JSON_ARRAY?: Record<string, unknown>;
+    XML?: Record<string, unknown>;
+    FORM?: Record<string, unknown>;
+  };
+  files?: Record<string, unknown>;
+  metadata?: Metadata;
+};
+
+type Metadata = {
+  sourceId: string;
+  workspaceId: string;
+  namespace: string;
+  instanceId: string;
+  sourceType: string;
+  sourceCategory: string;
+  trackingPlanId: string;
+  trackingPlanVersion: number;
+  sourceTpConfig: Object;
+  mergedTpConfig: Object;
+  destinationId: string;
+  jobRunId: string;
+  jobId: number;
+  sourceBatchId: string;
+  sourceJobId: string;
+  sourceJobRunId: string;
+  sourceTaskId: string;
+  sourceTaskRunId: string;
+  recordId: Object;
+  destinationType: string;
+  messageId: string;
+  oauthAccessToken: string;
+  messageIds: string[];
+  rudderId: string;
+  receivedAt: string;
+  eventName: string;
+  eventType: string;
+  sourceDefinitionId: string;
+  destinationDefinitionId: string;
+};
+
+type UserTransformationInput = {
+  VersionID: string;
+  ID: string;
+  Config: Object;
+};
+
+type DestinationDefinition = {
+  ID: string;
+  Name: string;
+  DisplayName: string;
+  Config: Object;
+  ResponseRules: Object;
+};
+
+type Destination = {
+  ID: string;
+  Name: string;
+  DestinationDefinition: DestinationDefinition;
+  Config: Object;
+  Enabled: boolean;
+  WorkspaceID: string;
+  Transformations: UserTransformationInput[];
+  RevisionID: string;
+};
+
+type UserTransformationLibrary = {
+  VersionID: string;
+};
+
+type ProcessorTransformRequest = {
+  request?: Object;
+  message: Object;
+  metadata: Metadata;
+  destination: Destination;
+  libraries: UserTransformationLibrary[];
+};
+
+type RouterTransformRequestData = {
+  request?: Object;
+  message: Object;
+  metadata: Metadata;
+  destination: Destination;
+};
+
+type RouterTransformRequest = {
+  input: RouterTransformRequestData[];
+  destType: string;
+};
+
+type ProcessorTransformResponse = {
+  output?: TransformedEvent | RudderMessage | Object;
```

Comment:
For User transform scenario we need this you can check UserTransform.ts under services for detailed info on high level in this module we are expecting a RudderMessage in place of transformed event in case of error we are returning object

This module is currently getting worked on by JC need to review if we can scope a update for this